src/PIL/Image.py

```diff
@@ -1862,7 +1847,7 @@ def paste(
         elif isinstance(im, tuple):
             source = im
         else:
-            source = cast(float, im)
+            source = im
```

Comment:
This `elif` could be removed now since it does the same thing as the `else` below it.

Reply:
You're right, I've updated the commit.